### PR TITLE
add boost 1.50

### PR DESCRIPTION
**Pull request by randy3k**
add boost 1.50, it is needed for some of the old software

### CODE REVIEW

File: boost150.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+  env :userpaths
+
+  option :universal
+  option "with-icu4c", "Build regexp engine with icu support"
+  option "without-single", "Disable building single-threading variant"
+  option "without-static", "Disable building static library variant"
+  option "with-mpi", "Build with MPI support"
+
+  deprecated_option "with-icu" => "with-icu4c"
```
Comment (DomT4) [inline boost150.rb]: Kill this line. We haven't deprecated anything here yet, since this is a new formula.

**Comment by DomT4** [comment]
One tiny nit, but that aside LGTM. Will merge it in after the nit is dealt with.

**Comment by randy3k** [comment]
I have removed the "deprecated" line.

**Comment by DomT4** [comment]
Thanks!

**Comment by DomT4** [comment]
Merged! Thanks for your contribution to Homebrew, we appreciate it! :smiley_cat:. For future reference the preferred commit message for new formulae is ` newformula123 1.2.3 (new formula) `, Thanks!

**Pull request closed without merging**